### PR TITLE
Document `federated-api` configuration

### PR DESCRIPTION
**Pull request by Ndpnt**

### CODE REVIEW

File: content/api/federated.en.md [hunk 1]
```diff
@@ -19,6 +19,71 @@ This API is offered as a preview, based on a first use case [defined](https://gi
...
+      "from": "The address from which to send the email",
+      "sendWarnings": "Boolean. Set to true to also send email in case of warning",
+    }
+  },
+}
+```
+
+The default configuration is merged with (and overridden by) environment-specific configuration that can be specified at startup with the `NODE_ENV` environment variable. See [node-config](https://github.com/node-config/node-config) for more information about configuration files.
+
+For developement, in order to have a local configuration that override exisiting config, it is recommended to create a `config/development.json` file with overridden values.
```
Comment (MattiSG) [inline content/api/federated.en.md:46]:
```suggestion
For development, in order to have a local configuration that overrides the existing config, it is recommended to create a `config/development.json` file.
```

File: content/api/federated.en.md [hunk 2]
```diff
@@ -19,6 +19,71 @@ This API is offered as a preview, based on a first use case [defined](https://gi
...
+
+### `collections` configuration key
+
+The `collections` configuration key is an array containing URLs and directly specified collections. Each item in the collections array can be either a URL pointing to a JSON file containing collections or a directly specified collection object.
```
Comment (MattiSG) [inline content/api/federated.en.md:50]:
```suggestion
The `collections` configuration key is an array containing URLs or specific collections. Each item in the collections array can be either a URL pointing to a JSON file containing Collections objects, or a Collection object.
```

File: content/api/federated.en.md [hunk 3]
```diff
@@ -19,6 +19,71 @@ This API is offered as a preview, based on a first use case [defined](https://gi
...
+
+#### Structure
+
+- **URLs**: URLs pointing to JSON files containing collections description. Each URL should respond with a JSON array containing collection objects.
+
+- **Directly specified collections**: Collection objects directly specified within the configuration. These collections must include properties: `name`, `id`, and `endpoint`.
```
Comment (MattiSG) [inline content/api/federated.en.md:56]:
```suggestion
- **Collection objects**: the description of a collection, in the form of an object with required properties: `name`, `id`, and `endpoint`.
```

File: content/api/federated.en.md [hunk 4]
```diff
@@ -19,6 +19,71 @@ This API is offered as a preview, based on a first use case [defined](https://gi
...
+
+```json
+{
+  …
+  {
+    "name": "The name of the collection",
+    "id": "A unique identifier for the collection",
```
Comment (MattiSG) [inline content/api/federated.en.md:63]:
```suggestion
    "id": "An identifier for the collection that must be unique across the whole federation",
```

File: content/api/federated.en.md [hunk 5]
```diff
@@ -19,6 +19,71 @@ This API is offered as a preview, based on a first use case [defined](https://gi
...
+}
+```
+
+#### Example
+
+```json
+"collections": [
+  "https://opentermsarchive.org/collections.json",
+  {
+    "name": "Direct Collection",
```
Comment (MattiSG) [inline content/api/federated.en.md:76]:
```suggestion
    "name": "My Collection",
```

File: content/api/federated.en.md [hunk 6]
```diff
@@ -19,6 +19,71 @@ This API is offered as a preview, based on a first use case [defined](https://gi
...
+    "id": "directCollection",
```
Comment (MattiSG) [inline content/api/federated.en.md:77]:
```suggestion
    "id": "mycollection",
```

File: content/api/federated.en.md [hunk 7]
```diff
@@ -19,6 +19,71 @@ This API is offered as a preview, based on a first use case [defined](https://gi
 
 The codebase for the Federated API is available on [`github.com/OpenTermsArchive/federated-api`](https://github.com/OpenTermsArchive/federated-api).
 
+## Configuring
+
+The default configuration can be found in `config/default.json`. The full reference is given below. You are unlikely to want to edit all of these elements.
+
+```js
+{
+  "port": "Port number on which the server will listen for incoming connections",
```
Comment (MattiSG) [inline content/api/federated.en.md:28]: > The default value should be provided.

File: content/api/federated.en.md [hunk 8]
```diff
@@ -19,6 +19,71 @@ This API is offered as a preview, based on a first use case [defined](https://gi
...
+  "collections": "Collections configuration key; see below",
```
Comment (MattiSG) [inline content/api/federated.en.md:29]:
```suggestion
  "collections": [  // Overriding this value creates a risk of splintering the federation, make sure to fully understand what happens when changing this value
    "List of collections to federate; see below for how to configure"
  ]
```

File: content/api/federated.en.md [hunk 9]
```diff
@@ -19,6 +19,71 @@ This API is offered as a preview, based on a first use case [defined](https://gi
 
 The codebase for the Federated API is available on [`github.com/OpenTermsArchive/federated-api`](https://github.com/OpenTermsArchive/federated-api).
 
+## Configuring
+
+The default configuration can be found in `config/default.json`. The full reference is given below. You are unlikely to want to edit all of these elements.
```
Comment (MattiSG) [inline content/api/federated.en.md:24]:
```suggestion
The default configuration can be found in `config/default.json`. The full reference is given below. In the vast majority of cases, the default values should be sufficient and only the email sending data should be specific.
```

File: content/api/federated.en.md [hunk 10]
```diff
@@ -19,6 +19,71 @@ This API is offered as a preview, based on a first use case [defined](https://gi
...
+    "smtp": {
+      "host": "SMTP server hostname", // Hostname of the SMTP server for sending emails
+      "username": "User for server authentication" // Password for server authentication is defined in environment variables, see the “Environment variables” section below
+    },
+    "sendMailOnError": { // Can be set to `false` if sending email on error is not needed
+      "to": "The address to send the email to in case of an error",
+      "from": "The address from which to send the email",
+      "sendWarnings": "Boolean. Set to true to also send email in case of warning",
+    }
+  },
```
Comment (MattiSG) [inline content/api/federated.en.md:40]:
```suggestion
  }
```

**Review by MattiSG — COMMENTED** [review]
Very good job! 👏 

I would be in favour of ordering the configuration entries from “most likely to edit” to “least likely to edit”.

**Review by clementbiron — APPROVED** [review]

**Review by MattiSG — APPROVED** [review]
👏

**Pull request merged**